ingestors/kafka/src/main/scala/hydra/kafka/programs/KeyAndValueSchemaV2Validator.scala

```diff
@@ -54,14 +55,17 @@ class KeyAndValueSchemaV2Validator[F[_]: Sync] private (schemaRegistry: SchemaRe
     resultOf(validators)
   }
 
-  private def validateRecordRecordTypeSchemas(schemas: Schemas, subject: Subject, streamType: StreamTypeV2): F[Unit] = {
+  private def validateRecordRecordTypeSchemas(schemas: Schemas, subject: Subject, streamType: StreamTypeV2, withRequiredFields: Boolean): F[Unit] = {
     val keyFields   = schemas.key.getFields.asScala.toList
     val valueFields = schemas.value.getFields.asScala.toList
 
     val validators = for {
       keyFieldsValidationResult                  <- validateKeyFields(keyFields)
       keySchemaEvolutionValidationResult         <- validateKeySchemaEvolution(schemas, subject)
       valueSchemaEvolutionValidationResult       <- validateValueSchemaEvolution(schemas, subject)
+      validateRequiredKeyFieldsResult            <- if(withRequiredFields) validateRequiredKeyFields(schemas.key, streamType) else Nil.pure
+      validateRequiredValueFieldsResult          <- if(withRequiredFields) validateRequiredValueFields(schemas.value, streamType) else Nil.pure
+      valueSchemaEvolutionValidationResult       <- validateValueSchemaEvolution(schemas, subject)
```

Comment:
Why do we need `validateValueSchemaEvolution(schemas, subject)` second time?

Reply:
Looks like it was accidentally added during merging process.